wpimath/src/main/java/edu/wpi/first/math/controller/PIDController.java

```diff
@@ -141,6 +144,18 @@ public void setD(double kd) {
     m_kd = kd;
   }
 
+  /**
+   * Sets the IZone range.
+   *
+   * @param izone izone range
+   */
+  public void setIZone(double izone) {
+    if (izone < 0) {
+      throw new IllegalArgumentException("IZone must be a non-zero positive number!");
```

Comment:
```suggestion
      throw new IllegalArgumentException("IZone must be a non-negative number!");
```
Zero is an allowed value (to disable)

Reply:
Zero would cause the integrator to continually reset. The value you want to disable i-zone is positive infinity.

Reply:
That's fine, that's the mathematically correct way. If that's what we want to do then this implementation needs to be changed to reflect that.

Reply:
REVLib implements their IZone such that 0 disables it. Do we prefer mathematical correctness or consistency with other libraries?

Reply:
Also, how would I do positive infinity for C++?

Reply:
I prefer doing the mathematically correct thing. You can use `std::numeric_limits<double>::infinity()` from the `<limits>` header.